agent/agentcontainers/containers_dockercli.go

```diff
@@ -0,0 +1,214 @@
+package agentcontainers
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coder/coder/v2/codersdk"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/xerrors"
+)
+
+// DockerCLILister is a ContainerLister that lists containers using the docker CLI
+type DockerCLILister struct{}
+
+var _ Lister = &DockerCLILister{}
+
+func (*DockerCLILister) List(ctx context.Context) (codersdk.WorkspaceAgentListContainersResponse, error) {
+	var stdoutBuf, stderrBuf bytes.Buffer
+	// List all container IDs, one per line, with no truncation
+	cmd := exec.CommandContext(ctx, "docker", "ps", "--all", "--quiet", "--no-trunc")
```

Comment:
Apparently I need to use `agentexec.Execer` here.

Reply:
Oh yeah, I forgot we had this package now. It's good we have linting for this, although in this instance I would almost argue `agentexec` isn't _strictly_ necessary as the agent is exec'ing Docker as an extension of its functionality.

**But** let's use `agentexec` anyway. At worst it might result in our exec of Docker not being granted high enough priority to finish its work in time, at best it protects us against Docker CLI going haywire and using too much memory/cpu, or something like that.